src/Tribe/Dates/Known_Range.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+
+class Tribe__Events__Dates__Known_Range {
+
+	/**
+	 * @var static
+	 */
+	protected static $instance;
+
+	/**
+	 * The class singleton constructor.
+	 *
+	 * @return Tribe__Events__Dates__Known_Range
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Determine the earliest start date and latest end date currently in the database
+	 * and store those values for future use.
+	 */
+	public function rebuild_known_range() {
+		global $wpdb;
+		remove_action( 'deleted_post', array( $this, 'rebuild_known_range' ) );
+
+		$earliest = strtotime( $wpdb->get_var( $wpdb->prepare( "
+				SELECT MIN(meta_value) FROM $wpdb->postmeta
+				JOIN $wpdb->posts ON post_id = ID
+				WHERE meta_key = '_EventStartDate'
+				AND post_type = '%s'
+				AND post_status IN ('publish', 'private', 'protected')
```

Comment:
Good idea, it was not supported on the original code but that's a good call. Notice I'm not using the stati when setting limit markers events to have a conservative approach: better to rebuild the known range on any possible limit event change than having to add an overlay of tracking limit events and stati.